File: python/shroomdk/models/compass/get_query_run.py
from typing import Dict, List, Optional, Union

from pydantic import BaseModel

from .core.query_run import QueryRun
from .core.rpc_request import RpcRequest
from .core.rpc_response import RpcResponse
from .core.tags import Tags


# Request
class GetQueryRunRpcRequestParams(BaseModel):
    queryRunId: str


class GetQueryRunRpcRequest(RpcRequest):
    method: str = "getQueryRun"
    params: List[GetQueryRunRpcRequestParams]


# Response
class GetQueryRunRpcResult(BaseModel):
    queryRun: QueryRun
    redirectedToQueryRun: Optional[QueryRun]


class GetQueryRunRpcResponse(RpcResponse):
    result: Union[GetQueryRunRpcResult, None]

File: js/src/types/compass/get-query-run.ts
import { QueryRun, RpcRequest, RpcResponse, BaseRpcRequest, BaseRpcResponse } from "./core";

// Request
export interface GetQueryRunRpcRequestParams {
  queryRunId: string;
}

export interface GetQueryRunRpcRequest extends RpcRequest<GetQueryRunRpcRequestParams> {
  method: "getQueryRun";
}

export class GetQueryRunRpcRequestImplementation
  extends BaseRpcRequest<GetQueryRunRpcRequestParams>
  implements GetQueryRunRpcRequest
{
  method: "getQueryRun" = "getQueryRun";
}

// Response
export interface GetQueryRunRpcResult {
  queryRun: QueryRun;
  redirectedToQueryRun?: QueryRun | null;
}

export interface GetQueryRunRpcResponse extends RpcResponse<GetQueryRunRpcResult> {}

export class GetQueryRunRpcResponseImplementation
  extends BaseRpcResponse<GetQueryRunRpcResult>
  implements GetQueryRunRpcResponse {}
